src/noteCard/NoteCard.tsx

```diff
@@ -1,36 +1,159 @@
+import { useState } from "react";
 import Card from "@mui/material/Card";
+import EditIcon from "@mui/icons-material/Edit";
 import CardHeader from "@mui/material/CardHeader";
-import CardMedia from "@mui/material/CardMedia";
 import CardContent from "@mui/material/CardContent";
-import { IconButton, Typography } from "@mui/material";
+import { Button, IconButton, TextField, Typography } from "@mui/material";
 import DeleteOutlineIcon from "@mui/icons-material/DeleteOutline";
 import { MockDataItem } from "src/types";
+import { Controller, SubmitHandler, useForm } from "react-hook-form";
+import { useUpdateBox } from "src/services/mutations/updateBox";
+import { useQueryClient } from "@tanstack/react-query";
+
 interface Props {
   item: MockDataItem;
   useDelete: (id?: string) => void;
+  useUpdate: (id?: string) => void;
 }
 
 const NoteCard = ({ item, useDelete }: Props) => {
+  const queryClient = useQueryClient();
+  const { mutate: updateBox } = useUpdateBox({
+    onSuccess: () => {
+      queryClient.invalidateQueries({ queryKey: ["boxes"] });
+      console.log("Box updated successfully");
+    },
+    onError: (error) => {
+      console.error("Error updating box:", error);
+    },
+  });
+
+  const { control, handleSubmit } = useForm<MockDataItem>({
+    defaultValues: item,
+  });
+  const [isEditing, setIsEditing] = useState<boolean>(true);
+
+  const onSubmit: SubmitHandler<MockDataItem> = (item) => {
+    updateBox(item);
+
+    setIsEditing((isEditing) => !isEditing);
+  };
+
+  // const modifedDimension = item.dimension
+  //   .map((dim) => dim + " x ")
+  //   .join("")
+  //   .slice(0, -2);
+
   return (
     <div>
-      <Card key={item._id}>
-        <CardHeader
-          action={
-            <IconButton onClick={() => useDelete(item._id)}>
-              <DeleteOutlineIcon />
-            </IconButton>
-          }
-          title={item.name}
-        />
-        <CardContent>
-          {/* <Typography variant="body2" color="textSecondary">
-            {item.dimension.map((val) => val + "  ")}
-          </Typography> */}
-          <Typography variant="body2" color="textSecondary">
-            {item.usage}
-          </Typography>
-        </CardContent>
-      </Card>
+      {isEditing ? (
+        <Card key={item._id}>
+          <CardHeader
+            action={
+              <>
+                <IconButton onClick={() => useDelete(item._id)}>
```

Comment:
Tutaj też możęmy uniknac inline funkcji. Lepiej zrobić tak:
onClick={prepareUseDelete(item._id));

a gdzieś wyżej: 
```
const prepareUseDelete = (id: string) => () => {
costamCostam(id);
}
```

Chodzi o to, że wywołujemy funkcję od razu przekazujac ID i ona nam zwraca kolejną funkcje, która już to ID pamięta:)